crates/iota-node/src/lib.rs

```diff
@@ -2209,6 +2222,61 @@ fn build_kv_store(
     )))
 }
 
+/// Builds and starts the gRPC server for the IOTA node based on the node's
+/// configuration.
+///
+/// This function performs the following tasks:
+/// 1. Checks if the node is a validator by inspecting the consensus
+///    configuration; if so, it returns early as validators do not expose gRPC
+///    APIs.
+/// 2. Checks if gRPC is enabled in the configuration.
+/// 3. Creates broadcast channels for checkpoint streaming.
+/// 4. Initializes the gRPC checkpoint service.
+/// 5. Spawns the gRPC server to listen for incoming connections.
+///
+/// Returns a tuple of optional broadcast channels for checkpoint summary and
+/// data.
+async fn build_grpc_server(
+    config: &NodeConfig,
+    state: Arc<AuthorityState>,
+    state_sync_store: RocksDbStore,
+) -> Result<(
+    Option<broadcast::Sender<Arc<CertifiedCheckpointSummary>>>,
+    Option<broadcast::Sender<Arc<CheckpointData>>>,
+)> {
+    // Validators do not expose gRPC APIs
+    if config.consensus_config().is_some() {
+        return Ok((None, None));
+    }
+
+    if !config.enable_grpc_api {
```

Comment:
combine both ifs with an `||`

Reply:
Thanks for pointing this out. I combined them at [de6cbc0](https://github.com/iotaledger/iota/pull/7511/commits/de6cbc0bc53de585d4ed2e7edde91b378274b282).